### ISSUE TITLE
Add support for `SHOW DATABASES LIKE`

### ISSUE THREAD

**Issue by ajm188** [vitessio/vitess#6315]
#### Overview of the Issue

When connected to a vtgate, running `SHOW DATABASES` shows all keyspaces the vtgate knows about. However when using `SHOW DATABASES LIKE '%some_pattern%'` the pattern is ignored completely, and the vtgate shows all keyspaces.

#### Reproduction Steps

Steps to reproduce this issue, example:

1. Create two keyspaces `foo` and `bar`
1. Run `SHOW DATABASES LIKE 'foo'`
1. View:

```
+----------------+
| Databases      |
+----------------+
| bar            |
| foo            |
+----------------+
```

#### Binary version

```sh
amason@workspace:~% vtgate --version
Version: 7200712c6 (Git branch 'HEAD') built on Wed May 13 09:48:15 PDT 2020 by amason@workspace using go1.14.2 linux/amd64
```

#### Operating system and Environment details

OS, Architecture, and any other information you can provide
about the environment.

```
amason@workspace:~% cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
amason@workspace:~% uname -r
4.4.0-1105-aws
amason@workspace:~% uname -m
x86_64
```

**Comment by klkyy2020** [comment]
database in vitess means keyspace  
`show database` command is same as `show vitess_keyspace`  
maybe we can add a table named keyspace in _vt database in mysql, and change `show databases like 'foo'` to `select * from _vt.keyspace where Database like 'foo'`;

**Comment by harshit-gangal** [comment]
Closed via #6750
